sureal/routine.py

```diff
@@ -75,12 +75,15 @@ def _get_reconstruction_stats(raw_scores, rec_scores):
 
     colors = ['black', 'gray', 'blue', 'red'] * 2
 
-    if dataset_filepath.endswith('.py'):
-        dataset = import_python_file(dataset_filepath)
-    elif dataset_filepath.endswith('.json'):
-        dataset = import_json_file(dataset_filepath)
+    if isinstance(dataset_filepath, list):
+        dataset = read_datasets_and_return_overlap(dataset_filepath)
```

Comment:
Instead of invoking `read_datasets_and_return_overlap` inside `run_subjective_models`, it is better to call `read_datasets_and_return_overlap` outside first, to produce a dataset_filepath (combined dataset written to a temp location), and then input to `run_subjective_models`. This way, the two functions are decoupled, and you won't have to increase the complexity of `run_subjective_models`.